### PR TITLE
Exporter mock sso logout

### PR DESCRIPTION
**Pull request by kevincarrogan**
### Aim

Allow logout to work on exporter SSO by clearing session data.

### CODE REVIEW

File: core/mock_sso/views.py [hunk 1]
```diff
@@ -147,7 +147,10 @@ def get(self, request, **kwargs):
 
 
 class Logout(RedirectView):
-    def get_redirect_url(self, *args, **kwargs):
+    def get_redirect_url(self):
```
Comment (depsiatwal) [inline core/mock_sso/views.py:150]: where we need fushing the session ?
isn't this what the real view does ?

File: core/mock_sso/views.py [hunk 2]
```diff
@@ -147,7 +147,10 @@ def get(self, request, **kwargs):
...
         redirect_url = self.request.build_absolute_uri("/")
-        self.request.session.flush()
         return redirect_url
+
+    def get(self, request, *args, **kwargs):
+        self.request.session.flush()
```
Comment (depsiatwal) [inline core/mock_sso/views.py:155]: ahh I see it does a redirect to here ?

**Review by depsiatwal — APPROVED** [review]
nice

File: core/mock_sso/views.py (hunk 1 above)
Comment (kevincarrogan) [reply core/mock_sso/views.py:150]: In my mind this method should only do what it's supposed to do without side-effects. In this case all it needs to do is return the URL that the `RedirectView` would redirect to.

File: core/mock_sso/views.py (hunk 2 above)
Comment (kevincarrogan) [reply core/mock_sso/views.py:155]: Nothing "redirects" here as such.

This is just how views work in that they get routed to the method that corresponds to the HTTP method being used to retrieve the page.

In this case I'm flushing the session here and then letting the parent (which does redirects) do its own thing.

**Pull request merged**